### ISSUE TITLE
How to specify extras to install optional-dependencies?

### ISSUE THREAD

**Issue by posutsai** [PyO3/maturin#604]
Hi,
I would like to install the dependencies specified in "optional-dependencies" in pyproject.toml as [pyO3-pure](https://github.com/PyO3/maturin/blob/d257dbbcd96874a28e3d9bb30c20df55732176e2/test-crates/pyo3-pure/pyproject.toml#L17). However, I don't know how to add `extras` argument in `maturin develop`. There is no related information in help message. Thank you.

**Comment by messense** [comment]
`maturin develop` doesn't support install optional dependencies yet. For now you can do `maturin build` and using pip to install the wheel to specify `extras` for example `pip install path/to/pyo3-pure.whl[test]'`

**Comment by Smurphy000** [comment]
I think this could be useful feature, any idea what the effort might be to add this ability?

**Comment by messense** [comment]
https://github.com/PyO3/maturin/blob/b556ed1d3a1eab65f180e5da5c00648173e77f1d/src/develop.rs#L50-L67

Just need to add optional dependencies here. The question is what to install since you can specify multiple extras with multiple dependencies in `pyproject.toml` for example:

```toml
[project.optional-dependencies]
test = [
  "pytest < 5.0.0",
  "pytest-cov[all]"
]
web = ["flask"]
```

**Comment by Smurphy000** [comment]
are you referring to test / web or that pytest-cov[all] is nested in test ?

**Comment by messense** [comment]
> are you referring to test / web or that pytest-cov[all] is nested in test ?

test / web

**Comment by messense** [comment]
Released in v0.11.4-beta.4: https://pypi.org/project/maturin/0.11.4_beta.4/#files , you can install it with `pip install -U --pre maturin`.
